codegenerator/cli/templates/static/codegen/src/EventProcessing.res

```diff
@@ -366,28 +312,26 @@ let processEventBatch = (
     }
 
     let elapsedTimeAfterDbWrite = timeRef->Hrtime.timeSince->Hrtime.toMillis->Hrtime.intFromMillis
-    let batchSize = eventBatch->Array.length
-    let handlerDuration = elapsedTimeAfterProcess - elapsedAfterLoad
-    let dbWriteDuration = elapsedTimeAfterDbWrite - elapsedTimeAfterProcess
+    let loaderDuration = elapsedTimeAfterLoaders
+    let handlerDuration = elapsedTimeAfterProcessing - loaderDuration
+    let dbWriteDuration = elapsedTimeAfterDbWrite - elapsedTimeAfterProcessing
     registerProcessEventBatchMetrics(
       ~logger,
-      ~batchSize,
-      ~loadDuration=elapsedAfterLoad,
+      ~loadDuration=loaderDuration,
       ~handlerDuration,
       ~dbWriteDuration,
-      ~latestProcessedBlocks,
     )
     if Env.Benchmark.shouldSaveData {
       Benchmark.addEventProcessing(
-        ~batchSize=eventBatch->Array.length,
+        ~batchSize,
         ~contractRegisterDuration=0,
-        ~loadDuration=elapsedAfterLoad,
+        ~loadDuration=loaderDuration,
```

Comment:
I guess we can get rid of contractRegisterDuration here?

Reply:
Yep